src/components/UI/docs/DocsLinks.tsx

```diff
@@ -24,16 +24,37 @@ interface Props {
 }
 
 export const DocsLinks: FC<Props> = ({ navLinks, toggleMobileAccordion }) => {
-  const { asPath, query: { slug } }  = useRouter();
+  const [openSections, setOpenSections] = useState<{ [key: string]: boolean }>({});
+  const {
+    asPath,
+    query: { slug }
+  } = useRouter();
+
+  useEffect(() => {
+    setOpenSections(
+      navLinks.reduce(
+        (acc, navLink) => ({
+          ...acc,
+          [navLink.id]: checkNavLinks({ items: navLink.items, pathCheck: asPath })
+        }),
+        {}
+      )
+    );
+  }, [asPath, navLinks]);
+
+  const handleSectionToggle = (id: string): void => {
+    setOpenSections(prev => ({ ...prev, [id]: !prev[id] }));
+  };
+
   return (
     <Stack border='2px' borderColor='primary'>
       {navLinks.map(({ id, to, items }, idx) => {
         const split = to?.split('/');
         const isActive = slug && split && split[split.length - 1] === slug[slug.length - 1];
-        const isSectionActive = checkNavLinks({ to, items, pathCheck: asPath.split('#')[0] })
+        const isSectionActive = openSections[id];
```

Comment:
If we're going to bother extracting this, let's just extract all of the logic
```suggestion
        const index = openSections[id] ? 0 : -1;
```

Reply:
Happy to ignore too and just get this in =)